apps/web/src/features/safenet/components/new-safe/SafenetReview/index.tsx

```diff
@@ -0,0 +1,151 @@
+import type { NewSafeFormData } from '@/components/new-safe/create'
+import { updateAddressBook } from '@/components/new-safe/create/logic/address-book'
+import { AppRoutes } from '@/config/routes'
+import { PayMethod } from '@/features/counterfactual/PayNowPayLater'
+import { SafeCreationEvent, safeCreationDispatch } from '@/features/counterfactual/services/safeCreationEvents'
+import { CF_TX_GROUP_KEY, replayCounterfactualSafeDeployment } from '@/features/counterfactual/utils'
+import SafenetLogo from '@/public/images/safenet/logo-safenet.svg'
+import { gtmSetChainId } from '@/services/analytics/gtm'
+import { useAppDispatch } from '@/store'
+import { useLazyDeploySafenetAccountQuery } from '@/store/safenet'
+import type { ReplayedSafeProps } from '@/store/slices'
+import CheckIcon from '@mui/icons-material/Check'
+import { Box, Typography } from '@mui/material'
+import { EMPTY_DATA, ZERO_ADDRESS } from '@safe-global/protocol-kit/dist/src/utils/constants'
+import { useRouter } from 'next/router'
+import type { Dispatch, SetStateAction } from 'react'
+import { useEffect } from 'react'
+import css from './styles.module.css'
+
+export type UseSubmitSafenetReviewHandlerProps = {
+  data: NewSafeFormData
+  onSubmit: (data: Partial<NewSafeFormData>) => void
+  setSubmitError: Dispatch<SetStateAction<string | undefined>>
+  setIsCreating: Dispatch<SetStateAction<boolean>>
+}
+
+export type SafenetReviewType = {
+  handleCreateSafeClick: () => void
+}
+
+export const useSubmitSafenetReviewHandler = ({
+  data,
+  onSubmit,
+  setSubmitError,
+  setIsCreating,
+}: UseSubmitSafenetReviewHandlerProps): SafenetReviewType => {
+  const router = useRouter()
+  const dispatch = useAppDispatch()
+  const [deploy, { data: deploymentData, error }] = useLazyDeploySafenetAccountQuery()
+
+  useEffect(() => {
+    if (!error) return
```

Comment:
See comment below: I think we should move this handling to the `handleSafeCreationClick` function or maybe even extract it into a new `safeCreationCallback` function that gets called with the result of the deployment call.